pyproject.toml

```
@@ -40,7 +40,7 @@ dependencies = [
     "z3-solver>=4.12,<4.14",
     "qecsim",
     "ldpc>=0.1.51",
-    "numpy>=1.26,<2; python_version > '3.11'", # some of our dependencies are not yet compatible with numpy 2.0
+    "numpy>=1.26,<3; python_version > '3.11'", # some of our dependencies are not yet compatible with numpy 2.0
     "numpy>=1.24,<2; python_version <= '3.11'", # some of our dependencies are not yet compatible with numpy 2.0
```

```suggestion
    "numpy>=1.26; python_version > '3.11'",
    "numpy>=1.24; python_version <= '3.11'",
```
Trying to drop the upper cap. Could be that it is really only the ldpc dependency that is not properly working with `numpy 2.0` yet. That needs another workaround anyway (which is in place right now)